dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordWriter.java

```diff
@@ -768,131 +762,67 @@ private void writeList(FieldDef fd, IRecordAccessor childRecordAccessor, List<Ob
      */
     private void writeDecimal(FieldDef fd, BigDecimal decimalValue)
     {
-        int precision = fd.getPrecision();
-        int scale = fd.getScale();
         int dataLen = (int) fd.getDataLen();
-
-        int mostSigDigit = precision - scale;
-        boolean isNegative = decimalValue.signum() == -1;
-        if (isNegative)
+        for (int i = 0; i < dataLen; i++)
         {
-            decimalValue = decimalValue.negate();
+            this.scratchBuffer[i] = 0;
```

Comment:
Yep, scratch buffer is larger than the max supported dataLen for a decimal, so that should be fine. Zeroing out the scratch buffer should only be required for decimal to do how I am constructing the values. Other types are setting bytes explicitly instead of ORing bits into the buffer